src/clj/puppetlabs/services/request_handler/request_handler_core.clj

```diff
@@ -259,21 +260,35 @@
     (= (:type x)
        :puppetlabs.services.request-handler.request-handler-core/bad-request)))
 
+(defn jruby-timeout?
+  [x]
+  "Determine if the supplied slingshot message is for a JRuby borrow timeout."
+  (when (map? x)
+    (= (:type x)
+       ::jruby/jruby-timeout)))
```

Comment:
Just to close the loop on this, @KevinCorcoran, @camlow325 and I were mystified for a bit how `::jruby/jruby-timeout` worked at all, we're pretty sure it's because `jruby` is aliased to the fully qualified namespace via `[puppetlabs.services.jruby.jruby-puppet-service :as jruby]`